packages/data/single-file/src/array-flatten.ts

```diff
@@ -0,0 +1 @@
+export const flatten = (x: any[]) => x.flat(Infinity);
```

Comment:
Add a type generic? We might want to add dts for nolyfill in the future.

Reply:
the origin type https://github.com/blakeembrey/array-flatten/blob/master/src/index.ts 

since it can nest to any level,(so can't just be `T[]`). I think copy the type definition from origin package maybe ok?